analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/Tests/VerifierTest.cs

```diff
@@ -159,6 +159,40 @@ public class Sample
         public void Verify_Cshtml() =>
             DummyWithLocationMapping.AddPaths("Dummy.cshtml").Verify();
 
+        [TestMethod]
+        public void Verify_Razor_WithFramework()
+        {
+            var verifierBuilder = DummyWithLocationMapping.AddPaths("Dummy.razor");
+            verifierBuilder.WithFramework("net6.0")
+                           .Invoking(x => x.Verify())
+                           .Should()
+                           .NotThrow();
+        }
+
+        [DataTestMethod]
+        [DataRow("net48")]
+        [DataRow("netcoreapp3.1")]
+        [DataRow("netstandard2.1")]
+        [DataRow("net5.0")]
+        public void Verify_Razor_WithFramework_NotSupported(string framework)
+        {
+            var verifierBuilder = DummyWithLocationMapping.AddPaths("Dummy.razor");
+            verifierBuilder.WithFramework(framework)
+                           .Invoking(x => x.Verify())
+                           .Should()
+                           .Throw<InvalidOperationException>().WithMessage("Razor compilation is supported only for .NET 6 and .NET 7 frameworks.");
+        }
+
+        [TestMethod]
+        public void Verify_Razor_ParseOptions()
```

Comment:
Since the `compile` method is public you can in the test acquire the compilation and check it for all sort of interesting things. Like the languageversion. Also that way you can assert that in case it needs to build with multiple languages it really does it. 
Also do some research on the compilation class, it might be that there is a way to check the targetframework.
What I saw is that we can basically check what targetframework specific assemblies we reference, and from that we can deduct if the correct targetframework was used. And at the same time there might be other better options.